test/test_framework.py

```diff
@@ -612,6 +612,63 @@ class CustomObject:
 
             framework_copy.commit()
 
+    def test_set_operations(self):
+        class SomeObject(Object):
+            state = StoredState()
+
+            def __init__(self, framework, key):
+                super().__init__(framework, key)
+
+        # (input, operation, (a, b) result, (b, a) result
```

Comment:
What do you think about having these be trailing comments on the first item in the list instead of a leader comment?  I think it would make it more obvious what each line was for without being too cluttered.  So, like:

```python
        test_operations = [(
            {"1"},  # other_set
            lambda a, b: a | b,  # operation
            {"1", "a", "b"},  # expected result of operation(obj, other_set)
            {"1", "a", "b"}  # expected result of operation(other_set, obj)
        ), (
```